spec.html

```diff
@@ -20977,6 +21317,21 @@ <h1>Static Semantics: Early Errors</h1>
             It is a Syntax Error if the BoundNames of |BindingList| contains any duplicate entries.
           </li>
         </ul>
+        <emu-grammar>
+          UsingDeclaration :
+            `using` BindingList `;`
+        </emu-grammar>
+        <ul>
+          <li>
+            It is a Syntax Error if the BoundNames of |BindingList| contains *"let"* or *"await"*.
+          </li>
+          <li>
+            It is a Syntax Error if the BoundNames of |BindingList| contains any duplicate entries.
+          </li>
+          <li>
+            It is a Syntax Error if the goal symbol is |Script| and |UsingDeclaration| is not contained, either directly or indirectly, within a |Block|, |CaseBlock|, |ForStatement|, |ForInOfStatement|, |FunctionBody|, |GeneratorBody|, |AsyncGeneratorBody|, |AsyncFunctionBody|, |ClassStaticBlockBody|, or |ClassBody|.
```

Comment:
@rbuckton What is an example of a `using` contained within a `ClassBody`? It can't be directly contained within one, and if indirectly contained, then it would already be directly contained within another production on the list (like ClassStaticBlockBody, FunctionBody, etc). During implementation we found the inclusion of ClassBody in this list confusing.

Reply:
You are correct, _ClassBody_ here is superfluous and should be removed.

Reply:
This should be addressed by https://github.com/tc39/proposal-explicit-resource-management/pull/220 and https://github.com/rbuckton/ecma262/pull/6